pkg/virt-api/rest/subresource.go

```diff
@@ -419,6 +423,28 @@ func (app *SubresourceAPIApp) RestartVMRequestHandler(request *restful.Request,
 	name := request.PathParameter("name")
 	namespace := request.PathParameter("namespace")
 
+	var gracePeriodint64 int64
+	var bodyMap map[string]string
+
+	if request.Request.Body != nil {
+		body, err := ioutil.ReadAll(request.Request.Body)
+		if err != nil {
+			response.WriteError(http.StatusForbidden, fmt.Errorf("Cannot read body from request, error: %s", err))
+			return
+		}
+		defer request.Request.Body.Close()
+		err = json.Unmarshal(body, &bodyMap)
```

Comment:
We normally allow yaml too. There should exist a `yaml.NewYamlOrJSONDecoder()` function which should help you with supporting both.

Reply:
Changed it to `NewYamlOrJSONDecoder` now